posts/banff-cohomology.md

```diff
@@ -0,0 +1,37 @@
+
+<!-- TEASER_END -->
+# Formalising cohomology theories
+
+Last week a group of formalisation enthusiasts, many of them Lean experts, met together at a [BIRS workshop](http://www.birs.ca/events/2023/5-day-workshops/23w5124) in Banff Canada, and discussed the formalisation of cohomology theories. Cohomology groups are linear invariants associated to typically nonlinear objects, and play an important role in several areas of mathematics. We list a few examples here: One can prove that two nonlinear objects are not isomorphic by showing that they have different cohomology. There are certain mathematical constructions that can be made (such as extending a map from a subgroup to a map from the whole group, or lifting a scheme over a base to a bigger base) if and only if some element of a cohomology group is zero, so if the entire cohomology group vanishes then the construction can always be made. The Hodge conjecture and the Tate conjecture are conjectures predicting that geometric resp. arithmetic behaviour of an appropriate mathematical object can be predicted from behaviour of its cohomology.
+
+There are many cohomology theories in the mathematical literature, and some have already been formalised in other theorem provers. But this is, in my mind, not good enough: we need a system which knows all cohomology theories and which furthermore knows all definitions of all cohomology theories, enabling us to do things such as writing down isomorphisms between cohomology theories, and giving us access to all the homological algebra (exact sequences, spectral sequences, derived functors,...) associated to cohomology theories. My personal aim for the conference was to understand where we are, and where we need to be, with regards to definitions and basic properties of as many cohomology theories as possible.
+
+Successes which had already been achieved before the conference were singular homology, which had been defined by Brendon Seamus Murphy, and group cohomology, which had been defined by Amelia Livingston. Amelia's work is already in Lean's mathematics library `mathlib`, and Brendon's is on the way. But just writing down the definition of a cohomology theory is *not good enough*. One needs to prove theorems using the cohomology theory, to demonstrate that the definition is actually usable in practice (for example, one could imagine developing some huge abstract theory of derived functors, and then defining many cohomology theories as derived functors, and then being completely unable to compute the cohomology of anything at all). In his talk, Brendon outlined his proof of the Brouwer fixed point theorem using singular homology, and also his computation of the homology groups of spheres. Similarly, Amelia explained how she had proved the long exact sequence and inflation-restriction sequence for group cohomology. These are good signs: it means that the definitions are usable.
```

Comment:
The *definition* of singular homology in Lean should not be attributed to Brendan, as I understand it -- it has been around for years (in a basically unworkable form), see
https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Ext.20and.20Tor/near/246037632
Rather, Brendan was the first person to *use* it.

Reply:
I have slightly rewritten this para (and corrected the spelling of Brendan's name!)